pkg/registry/pod/etcd/etcd.go

```diff
@@ -142,7 +143,14 @@ func (r *BindingREST) New() runtime.Object {
 // Create ensures a pod is bound to a specific host.
 func (r *BindingREST) Create(ctx api.Context, obj runtime.Object) (out runtime.Object, err error) {
 	binding := obj.(*api.Binding)
-	err = r.assignPod(ctx, binding.PodID, binding.Host)
+	// TODO: move me to a binding strategy
+	if len(binding.Target.Kind) != 0 && (binding.Target.Kind != "Node" && binding.Target.Kind != "Minion") {
```

Comment:
I think it's

* kubelet needs to validate constraints and report failed
* scheduler / rc need to handle failed correctly
* we remove the atomic check and replace it with a query
* we remove setting the bindings and replace it with writing status.Host

----- Original Message -----
> > @@ -142,7 +143,14 @@ func (r *BindingREST) New() runtime.Object {
> >  // Create ensures a pod is bound to a specific host.
> >  func (r *BindingREST) Create(ctx api.Context, obj runtime.Object) (out
> >  runtime.Object, err error) {
> >  	binding := obj.(*api.Binding)
> > -	err = r.assignPod(ctx, binding.PodID, binding.Host)
> > +	// TODO: move me to a binding strategy
> > +	if len(binding.Target.Kind) != 0 && (binding.Target.Kind != "Node" &&
> > binding.Target.Kind != "Minion") {
> 
> How hard would it be to eliminate the separate etcd object and just
> write/read the pod? To make the separate object work, we really need atomic
> deletion of the pod and binding (or a background reconciliation loop, which
> I'd like to avoid).
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5054/files#r25899370